### ISSUE TITLE
to_string(cart_coords=True) broken in GaussianInput

### ISSUE THREAD

**Issue by ConradJohnston** [materialsproject/pymatgen#2863]
**Describe the bug**
For the input generator class for Gaussian:
[pymatgen/io/gaussian.py : GaussianInput 
](https://github.com/materialsproject/pymatgen/blob/0d2062a01bf0ccb36cb736c9a5d139787c55e4c1/pymatgen/io/gaussian.py#LL87C14-L87C14)
the `to_string()` method should produce a string consisting of a Gaussian input. 
The option "`cart_coords=True`" is supposed to write cartesian coordinates into the input, while the default value of false writes a Z-matrix.

Choosing cartesian coordinates leads to the error:
`TypeError: str.join() takes exactly one argument (2 given)`

The buggy line is here:
https://github.com/materialsproject/pymatgen/blob/40cbf1d7c47bb2eb75509780e356c9877ae9c2f7/pymatgen/io/gaussian.py#L423

It looks to have been broken recently on Jan 6 in commit 0d2062a .

**To Reproduce**

```
from pymatgen.core import Molecule
coords = [[0.000000, 0.000000, 0.000000],
          [0.000000, 0.000000, 1.089000],
          [1.026719, 0.000000, -0.363000],
          [-0.513360, -0.889165, -0.363000],
          [-0.513360, 0.889165, -0.363000]]
methane = Molecule(["C", "H", "H", "H", "H"], coords)

from pymatgen.io import gaussian
inp = gaussian.GaussianInput(mol=methane)
inp.to_string(cart_coords=True)
```

**Comment by gVallverdu** [comment]
Seems to be fixed from commit  https://github.com/materialsproject/pymatgen/commit/577b2cddf8d8cbad218d7285341ddf5869d2cdea

Will be included in the next version.

**Comment by ConradJohnston** [comment]
@gVallverdu  - great!
See also #2865

**Comment by janosh** [comment]
Fixed in #2891.
